src/CanvasItem/Rect.js

```diff
@@ -0,0 +1,189 @@
+    const onMove = e => {
+      if (!isMouseDown.current) return; // patch: fix windows press win key during mouseup issue
+      e.stopPropagation();
+      const { clientX, clientY } = e;
+      const rotateVector = {
+        x: clientX - center.x,
+        y: clientY - center.y
+      };
+      const angle = getAngle(startVector, rotateVector);
+      onRotate(angle, startAngle);
+    };
+    const onUp = () => {
+      document.removeEventListener("mousemove", onMove);
+      document.removeEventListener("mouseup", onUp);
+      if (!isMouseDown.current) return;
+      isMouseDown.current = false;
+    };
+    document.addEventListener("mousemove", onMove);
+    document.addEventListener("mouseup", onUp);
+  };
+
+  // Resize
+  const startResize = (e, cursor, type) => {
+    document.body.style.cursor = cursor;
+    e.stopPropagation();
+    const { clientX: startX, clientY: startY } = e;
+    const rect = {
+      centerX,
+      centerY,
+      height,
+      rotateAngle,
+      width
+    };
+    isMouseDown.current = true;
+    const onMove = e => {
+      if (!isMouseDown.current) return; // patch: fix windows press win key during mouseup issue
+      e.stopPropagation();
+      const { clientX, clientY } = e;
+      const deltaX = clientX - startX;
+      const deltaY = clientY - startY;
+      const alpha = Math.atan2(deltaY, deltaX);
+      const deltaL = getLength(deltaX, deltaY);
+      const isShiftKey = e.shiftKey;
+      onResize(deltaL, alpha, rect, type, isShiftKey);
+    };
+
+    const onUp = () => {
+      document.body.style.cursor = "auto";
+      document.removeEventListener("mousemove", onMove);
+      document.removeEventListener("mouseup", onUp);
+      if (!isMouseDown.current) return;
+      isMouseDown.current = false;
+    };
+    document.addEventListener("mousemove", onMove);
+    document.addEventListener("mouseup", onUp);
+  };
+
+  const startDrag = e => {
+    const { clientX: startX, clientY: startY } = e;
+    const startCenterX = centerX;
+    const startCenterY = centerY;
+
+    isMouseDown.current = true;
+    e.stopPropagation();
+
+    const onMove = e => {
+      if (!isMouseDown.current) return; // patch: fix windows press win key during mouseup issue
+      e.stopPropagation();
+      const { clientX, clientY } = e;
+      const deltaX = clientX - startX;
+      const deltaY = clientY - startY;
+      const newCenterX = startCenterX + deltaX;
+      const newCenterY = startCenterY + deltaY;
+      onDrag(newCenterX, newCenterY);
+    };
+    const onUp = () => {
+      document.removeEventListener("mousemove", onMove);
+      document.removeEventListener("mouseup", onUp);
+      if (!isMouseDown.current) return;
+      isMouseDown.current = false;
+    };
+    document.addEventListener("mousemove", onMove);
+    document.addEventListener("mouseup", onUp);
+  };
+
+  const direction = zoomable
+    .split(",")
+    .map(d => d.trim())
+    .filter(d => d); // TODO: may be speed up
+
```

Comment:
Lack of comments and lots of code just grouped together with no new lines. Very hard to read.